src/crypto/test/digest_bench.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "crypto/hash.h"
+
+#define PICOBENCH_IMPLEMENT_WITH_MAIN
+#include <picobench/picobench.hpp>
+
+static void evercrypt_sha256(picobench::state& s)
+{
+  std::vector<uint8_t> v(s.iterations());
+  for (size_t i = 0; i < v.size(); ++i)
+  {
+    v.data()[i] = rand();
+  }
+
+  crypto::Sha256Hash h;
+
+  s.start_timer();
+  for (size_t i = 0; i < 1000; ++i)
```

Comment:
Is the performance difference as stark if we don't repeat this 1000 times? I think we care more about the single-iteration speed than hot-cache looping.

Reply:
I don't know, but it's guaranteed to be much less stable and therefore much harder to compare. We do hash a fair bit in the merkle tree, and this is only going to increase with multi-label receipts, so while we won't do 1000 in a row, we'll do more than the odd one shot digest.

Reply:
The answer seems to be yes, by the way:

```
amchamay@amchamay2:/data/amchamay/CCF/build$ ./digest_bench 
SHA-256:
===============================================================================
   Name (baseline is *)   |   Dim   |  Total ms |  ns/op  |Baseline| Ops/second
===============================================================================
evercrypt_digest_sha256 * |     128 |     0.001 |       7 |      - |142222222.2
    mbedtls_digest_sha256 |     128 |     0.001 |       5 |  0.778 |182857142.9
evercrypt_digest_sha256 * |     512 |     0.003 |       5 |      - |196923076.9
    mbedtls_digest_sha256 |     512 |     0.002 |       3 |  0.769 |256000000.0
evercrypt_digest_sha256 * |    8192 |     0.034 |       4 |      - |243086053.4
    mbedtls_digest_sha256 |    8192 |     0.026 |       3 |  0.760 |320000000.0
evercrypt_digest_sha256 * |  131072 |     0.534 |       4 |      - |245497317.9
    mbedtls_digest_sha256 |  131072 |     0.402 |       3 |  0.754 |325804182.4
evercrypt_digest_sha256 * |  524288 |     2.133 |       4 |      - |245808086.2
    mbedtls_digest_sha256 |  524288 |     1.614 |       3 |  0.757 |324794807.1
===============================================================================
```